controllers/ebpf/agent_controller.go

```diff
@@ -493,12 +461,40 @@ func (c *AgentController) configureFlowFilter(filter *flowslatest.EBPFFlowFilter
 			}
 		}
 	}
-
+	if filter.SourcePorts.Type == intstr.String {
+		config = append(config, corev1.EnvVar{Name: envFilterSourcePortRange,
+			Value: filter.SourcePorts.String(),
+		})
+	}
+	if filter.SourcePorts.Type == intstr.Int {
+		config = append(config, corev1.EnvVar{Name: envFilterSourcePort,
+			Value: strconv.Itoa(filter.SourcePorts.IntValue()),
+		})
+	}
+	if filter.DestPorts.Type == intstr.String {
+		config = append(config, corev1.EnvVar{Name: envFilterDestPortRange,
+			Value: filter.DestPorts.String(),
+		})
+	}
+	if filter.DestPorts.Type == intstr.Int {
+		config = append(config, corev1.EnvVar{Name: envFilterDestPort,
+			Value: strconv.Itoa(filter.DestPorts.IntValue()),
+		})
+	}
+	if filter.Ports.Type == intstr.String {
+		config = append(config, corev1.EnvVar{Name: envFilterPortRange,
+			Value: filter.Ports.String(),
+		})
+	}
+	if filter.Ports.Type == intstr.Int {
+		config = append(config, corev1.EnvVar{Name: envFilterPort,
+			Value: strconv.Itoa(filter.Ports.IntValue()),
+		})
+	}
```

Comment:
this won't work from ebpf agent processing the code look for the protocol then check the ports so if protocol not set it won't try to match the ports

Reply:
Can't we catch all protocols in that case ? :thinking: 
I thought that was the output of our conversations on that one

If not, I'll fall back on your suggestion about an error in FlowCollector status

Reply:
we can't make a guess about the protocol when we parse the pkt in ebpf so not sure how will the catch all works lets chat about it so we both are clear ?

Reply:
looks now I updated the agent to work in this mode pls bring this fix to 1.6.1